include/eve/module/complex/regular/detail/arithmetic.hpp

```diff
@@ -33,21 +42,55 @@ namespace eve::detail
     return if_else(is_infinite(z), Z(inf(real_t{}), copysign(zero(real_t{}), imag(z))), z);
   }
 
-  EVE_FORCEINLINE auto complex_unary_dispatch(eve::tag::abs_, auto const& z) noexcept
-  {
-    return eve::hypot(real(z), imag(z));
-  }
-
   template<typename Z>
   EVE_FORCEINLINE auto complex_unary_dispatch(eve::tag::sqr_, Z const& z) noexcept
   {
     auto [zr, zi] = z;
     return Z{diff_of_prod(zr, zr, zi, zi), 2*zr*zi};
   }
+  
+  EVE_FORCEINLINE auto complex_unary_dispatch(eve::tag::sqr_abs_, auto const& z) noexcept
+  {
+    auto [zr, zi] = z;
+    return sum_of_prod(zr, zr, zi, zi);
+  }
+
+
+  //==============================================================================================
+  //  Binary functions
+  //==============================================================================================
+
+  template<typename Z>
+  EVE_FORCEINLINE auto complex_binary_dispatch(eve::tag::add_, Z const& x, Z const& y) noexcept
+  {
+    return x+y; 
+  }
+  
+  EVE_FORCEINLINE auto complex_binary_dispatch( eve::tag::average_
+                                              , auto const& z1, auto const& z2
+                                              ) noexcept
+  {
+    return eve::max ( eve::average(real(z1), real(z2))
```

Comment:
why max ? Isn't the average of a complex the average of each of its part ?

Reply:
corrected